services/processor/native/call.go

```diff
@@ -77,45 +84,51 @@ func (s *service) processMethodCall(contractInfo *types.ContractInfo, methodInfo
 	}
 
 	// get contract error
-	outErr, err := s.createContractError(methodInfo, outValues[len(outValues)-1])
+	outErr, err = s.createContractError(methodInfo, outValues[len(outValues)-1])
 	return outArgs, outErr, err
 }
 
-func (s *service) verifyMethodArgs(methodInfo *types.MethodInfo, implementation interface{}, args []*protocol.MethodArgument) ([]reflect.Value, error) {
+func (s *service) verifyMethodArgs(ctx types.Context, methodInfo *types.MethodInfo, implementation interface{}, args []*protocol.MethodArgument) ([]reflect.Value, error) {
 	res := []reflect.Value{}
 	methodType := reflect.ValueOf(implementation).Type()
-	if methodType.NumIn()-1 != len(args) {
-		return nil, errors.Errorf("method '%s' takes %d args but received %d", methodInfo.Name, methodType.NumIn()-1, len(args))
+	if methodType.NumIn() < 2 || methodType.In(1) != reflect.TypeOf(ctx) {
```

Comment:
`2` being a magic number might be better as extracted to a constant

Reply:
done